mantidproject/mantid
# GSASIIRefineFitPeaks reintegrated with latest version of GSAS-II

The algorithm `GSASIIRefineFitPeaks`  has been brought up to date and integrated with the most recent version of GSAS-II. The single peak fitting functionality has been disabled for the time being, as the new scriptable API for GSAS-II does not support it.

More functionality is expected to be added to the algorithm as scientists' specific fitting needs become apparent, but this is a good starting block for bringing support for engineering back into Mantid.

**To test:**

You'll need a fairly recent version of GSAS-II (later than April 2017) - you can get it [here](https://subversion.xray.aps.anl.gov/trac/pyGSAS). Some example data is [here](https://github.com/mantidproject/mantid/files/1397963/GSASIIRefineFitPeaks_data.zip)

The following script, as a minimum should give you some data:
```
ws = Load(Filename=<path to data>/"focused_bank1_ENGINX00256663.nxs")

GOF, Rwp, lattice_tbl = GSASIIRefineFitPeaks(PathToGSASII=<path to gsas>, 
                                             SaveGSASIIProjectFile=<some new GSAS-II project>,
                                             PhaseInfoFile=<path to dath>/"FE_ALPHA.cif",
                                             InstrumentFile=<path to data>/"template_ENGINX_241391_236516_North_bank.prm">,
                                             InputWorkspace=ws,
                                             RefinementMethod="Rietveld refinement")                                                          
print "Goodness of fit coefficient: {0:.5f}".format(GOF)
print "Weighted profile R-factor (Rwp): {0:.5f}".format(Rwp)
print ("Lattice parameters, a: {a}, b: {b}, c: {c}, alpha: {alpha}, beta: {beta}, gamma: {gamma}, "
         "Volume: {volume:.3f}".format(**lattice_tbl.row(0)))
```

Try setting `RefinementMethod="Pawley refinement"`. The unit tests won't run on the build servers, as they are skipped if GSAS-II is not found, so try running those on a local machine too (you need to have GSAS-II in a directory called `g2conda` (the installation default) within 2 levels of your root directory.

Also code review, as the algorithm has been completely rewritten and will be added to, so I want to make sure I haven't shot myself (or anyone else) in the foot without noticing early on.

Fixes #19578

**Release Notes** 

[Here](https://github.com/mantidproject/mantid/commit/bb3ce4705b2dd04bfa5247de41b20015839f44d9)

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
